### PR TITLE
Add ifMatch options to destroyById

### PR DESCRIPTION
**Pull request by aenario**

### CODE REVIEW

**Comment by aenario** [comment]
Looks like this requires the same change  to be made in cozy-stack.

File: src/files.js [hunk 1]
```diff
@@ -307,8 +307,12 @@ export function restoreById (cozy, id) {
   return cozyFetchJSON(cozy, 'POST', `/files/trash/${encodeURIComponent(id)}`)
 }
 
-export function destroyById (cozy, id) {
-  return cozyFetchJSON(cozy, 'DELETE', `/files/trash/${encodeURIComponent(id)}`)
+export function destroyById (cozy, id, opts) {
+  return cozyFetchJSON(cozy, 'DELETE', `/files/trash/${encodeURIComponent(id)}`, undefined, {
+    headers: {
+      'If-Match': (opts && opts.ifMatch) ? opts.ifMatch : ''
```
Comment (nono) [inline src/files.js]: Can we simplify this line to this?

```js
'If-Match': opts && opts.ifMatch
```

File: test/integration/files.js [hunk 2]
```diff
@@ -196,6 +196,24 @@ describe('files API', async function () {
...
+    const created = await cozy.client.files.create('datastring2', {
+      name: 'foo_' + random(),
+      contentType: 'application/json',
+      lastModifiedDate: new Date('Wed, 01 Feb 2017 10:24:42 GMT')
+    })
+    await cozy.client.files.trashById(created._id)
+    await cozy.client.files.destroyById(created._id, {ifMatch: 'badbeef'})
+      .then(
+        () => { throw new Error('should reject') },
+        (err) => { err.should.be.an.Error })
```
Comment (sebn) [inline test/integration/files.js]: I didn't know this was possible!

File: src/files.js (hunk 1 above)
Comment (aenario) [inline src/files.js]: Unfortunately not. In chrome `fetch(..., {headers: {"If-Match": undefined}})` send an `If-Match: undefined` header

Comment (sebn) [inline src/files.js]: `'IF-Match': (opts && opts.ifMatch) || ''`?

**Comment by sebn** [comment]
👍

File: test/integration/files.js [hunk 3]
```diff
@@ -196,6 +196,24 @@ describe('files API', async function () {
     trashed.should.have.length(0)
   })
 
+  it('does not destroy a trashed file or directory with wrong sum', async function () {
```
Comment (sebn) [inline test/integration/files.js]: `s/sum/rev`?

File: test/integration/files.js [hunk 4]
```diff
@@ -196,6 +196,24 @@ describe('files API', async function () {
...
+      name: 'foo_' + random(),
+      contentType: 'application/json',
+      lastModifiedDate: new Date('Wed, 01 Feb 2017 10:24:42 GMT')
+    })
+    await cozy.client.files.trashById(created._id)
+    await cozy.client.files.destroyById(created._id, {ifMatch: 'badbeef'})
+      .then(
+        () => { throw new Error('should reject') },
+        (err) => { err.should.be.an.Error })
+    let trashed = await cozy.client.files.listTrash()
```
Comment (sebn) [inline test/integration/files.js]: `const`?

File: src/files.js (hunk 1 above)
Comment (aenario) [inline src/files.js]: Finally, I copied how it was done in other functions with an intermediary ifMatch variable.

**Pull request merged**